### PR TITLE
[WIP] [SMTChecker] Special variables (block, msg, tx, now)

### PR DESCRIPTION
**Pull request by leonardoalt**
Depends on #5189 #5209

### CODE REVIEW

File: libsolidity/formal/SpecialVariables.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace dev
+{
+namespace solidity
+{
+
+class MemberAccess;
+
+/**
+ * This class encodes the special variables block.*, msg.*, tx.* and now.
```
Comment (chriseth) [inline libsolidity/formal/SpecialVariables.h:32]: gasleft?

File: libsolidity/formal/SpecialVariables.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	void resetMember(MemberAccess const& _memberAccess);
+
+	smt::Expression operator[](std::string const& _member) const
+	{
+		return m_ssaVariables.at(_member)();
+	}
+
+	smt::Expression operator[](MemberAccess const& _memberAccess)
+	{
+		ASTString member = _memberAccess.memberName();
```
Comment (chriseth) [inline libsolidity/formal/SpecialVariables.h:52]: const&

**Comment by chriseth** [comment]
I somehow have the impression that this might be overly complicated code, but I also cannot suggest any specific improvements :(

**Comment by leonardoalt** [comment]
I see that too. I'm also refactoring types, so it might be easier to just do it inside `SMTChecker` when the refactoring is done.

**Comment by leonardoalt** [comment]
Closing in favor of #5272

**Pull request closed without merging**